### PR TITLE
schedule, get route patterns

### PR DESCRIPTION
**Pull request by ryan-mahoney**
#### Summary of changes
**Asana Ticket:** [⏱Schedules | Directionality - Get trip pattern data](https://app.asana.com/0/555089885850811/1127979261717805)

- get all route patterns for a route, grouped by direction
- get all shapes for route, organized in a map by shape.id
- allow shape repo call to not require a `direction_id`
- remove some file that appeared to have been checked-in accidentally

<br>
Assigned to: @phildarnowsky

### CODE REVIEW

File: apps/routes/lib/repo.ex [hunk 1]
```diff
@@ -47,11 +47,13 @@ defmodule Routes.Repo do
...
-  @spec get_shapes(String.t(), 0 | 1, boolean) :: [Shape.t()]
-  def get_shapes(route_id, direction_id, filter_negative_priority? \\ true) do
+  @spec get_shapes(String.t(), Keyword.t(), boolean) :: [Shape.t()]
+  def get_shapes(route_id, opts, filter_negative_priority? \\ true) do
+    opts = Keyword.put(opts, :route, route_id)
+
     shapes =
-      cache({route_id, direction_id}, fn _ ->
-        case Shapes.all(route: route_id, direction_id: direction_id) do
+      cache(opts, fn _ ->
```
Comment (gshkolnik) [inline apps/routes/lib/repo.ex]: are the same `opts` always going to be in the same order? If not, we might be hurting our cache hit rate.

Comment (phildarnowsky) [reply apps/routes/lib/repo.ex]: Easy to fix by changing `opts` to `Enum.sort(opts)`

File: apps/route_patterns/lib/repo.ex [hunk 2]
```diff
@@ -10,6 +10,7 @@ defmodule RoutePatterns.Repo do
       nil -> [route: route_id]
       direction_id -> [route: route_id, direction_id: direction_id]
     end
+    |> Keyword.put(:sort, "sort_order")
```
Comment (phildarnowsky) [inline apps/route_patterns/lib/repo.ex]: Is this what you were referring to earlier when you said the default sort order seemed to be by typicality?

File: apps/routes/lib/repo.ex [hunk 3]
```diff
@@ -47,11 +47,13 @@ defmodule Routes.Repo do
     end
   end
 
-  @spec get_shapes(String.t(), 0 | 1, boolean) :: [Shape.t()]
-  def get_shapes(route_id, direction_id, filter_negative_priority? \\ true) do
+  @spec get_shapes(String.t(), Keyword.t(), boolean) :: [Shape.t()]
+  def get_shapes(route_id, opts, filter_negative_priority? \\ true) do
```
Comment (phildarnowsky) [inline apps/routes/lib/repo.ex]: so the idea is that, now that `direction_id` is optional, we can call `get_shapes` without it to get all shapes in both directions for the route in question when we fetch all the data for this new feature?

File: apps/site/lib/site_web/controllers/schedule/line.ex [hunk 4]
```diff
@@ -76,25 +89,63 @@ defmodule SiteWeb.ScheduleController.Line do
...
+
+    Map.put(route_pattern, :shape_id, shape_id)
+  end
+
+  @spec get_route_shape_map(Route.id_t()) :: map
+  def get_route_shape_map(route_id) do
+    route_id
+    |> get_route_shapes()
+    |> Enum.map(fn shape -> {shape.id, shape} end)
+    |> Map.new()
```
Comment (gshkolnik) [inline apps/site/lib/site_web/controllers/schedule/line.ex]: no need to do `Enum.map`, `Map.new` does that already.

File: apps/route_patterns/lib/repo.ex (hunk 2 above)
Comment (ryan-mahoney) [reply apps/route_patterns/lib/repo.ex]: yes

File: apps/routes/lib/repo.ex (hunk 3 above)
Comment (ryan-mahoney) [reply apps/routes/lib/repo.ex]: yes

**Pull request merged**